lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.PlantLoop.rb

```diff
@@ -135,114 +173,367 @@ def plant_loop_apply_prm_number_of_chillers(plant_loop, sizing_run_dir = nil)
       chiller_compressor_type = 'Centrifugal'
     end
 
-    # Get all existing chillers and pumps
-    chillers = []
-    pumps = []
-    plant_loop.supplyComponents.each do |sc|
-      if sc.to_ChillerElectricEIR.is_initialized
-        chillers << sc.to_ChillerElectricEIR.get
-      elsif sc.to_PumpConstantSpeed.is_initialized
-        pumps << sc.to_PumpConstantSpeed.get
-      elsif sc.to_PumpVariableSpeed.is_initialized
-        pumps << sc.to_PumpVariableSpeed.get
+    if chillers.length > num_chillers
+      OpenStudio.logFree(OpenStudio::Error, 'prm.log', "For #{plant_loop.name}, the existing number of chillers exceeds the recommended amount. We have not accounted for this in the codebase yet.")
+    end
+
+    # Determine the per-chiller capacity and sizing factor
+    per_chiller_sizing_factor = (1.0 / num_chillers).round(2)
+    per_chiller_cap_w = cap_w / num_chillers
+
+    # Set the sizing factor and the chiller types
+    # chillers.each_with_index do |chiller, i|
+    for i in 0..num_chillers - 1
+      # if not enough chillers exist, create a new one. Else reference the i'th chiller
+      if i <= chillers.length - 1
+        chiller = chillers[i]
+      else
+        chiller = OpenStudio::Model::ChillerElectricEIR.new(model)
+        plant_loop.addSupplyBranchForComponent(chiller)
+        chiller.setReferenceLeavingChilledWaterTemperature(dsgn_sup_wtr_temp_c)
+        chiller.setLeavingChilledWaterLowerTemperatureLimit(OpenStudio.convert(36.0, 'F', 'C').get)
+        chiller.setReferenceEnteringCondenserFluidTemperature(OpenStudio.convert(95.0, 'F', 'C').get)
+        chiller.setMinimumPartLoadRatio(0.15)
+        chiller.setMaximumPartLoadRatio(1.0)
+        chiller.setOptimumPartLoadRatio(1.0)
+        chiller.setMinimumUnloadingRatio(0.25)
+        chiller.setChillerFlowMode('ConstantFlow')
+        chiller.setReferenceCOP(default_cop)
+
+        condenser_water_loop.get.addDemandBranchForComponent(chiller) if condenser_water_loop.is_initialized
+
       end
+
+      chiller.setName("#{template} #{chiller_cooling_type} #{chiller_compressor_type} Chiller #{i + 1} of #{num_chillers}")
+      chiller.setSizingFactor(per_chiller_sizing_factor)
+      chiller.setReferenceCapacity(per_chiller_cap_w)
+      chiller.setCondenserType(chiller_cooling_type)
+      chiller.additionalProperties.setFeature('compressor_type', chiller_compressor_type)
+
+      # Add inlet pump
+      new_pump = OpenStudio::Model::PumpVariableSpeed.new(plant_loop.model)
+      new_pump.setName("#{chiller.name} Inlet Pump")
+      new_pump.setRatedPumpHead(original_pump.ratedPumpHead / num_chillers)
+      new_pump.setCoefficient1ofthePartLoadPerformanceCurve(original_pump.coefficient1ofthePartLoadPerformanceCurve)
```

Comment:
If we need to use variable speed pump, since the pump should be a constant speed pump, we should either 1) use the coefficient of a pump "riding" the curves (they're already used in the code base) or 2) have all coefficients be 0 except the constant one to not model and part load improvement.

Reply:
That's a great point and one I missed when refactoring this. I should update this.

Reply:
Should we had a note in the code to document the choice?

Reply:
Looks like this was resolved.